Feat/eb sf batch

srinivasreddych opened the pull request:
Fixed #40 

This pattern demonstrates AWS EventBridge Cron based invoking Step Fucntions invoking AWS Batch jobs


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

dgraeber commented on modules/examples/eb-sf-batch/app.py:
```diff
@@ -0,0 +1,53 @@
+)
+
+# CfnOutput(
+#     scope=stack,
```
Why are we not exporting things like the ECR repo and the Step Function ARN?

srinivasreddych replied:
I can export them. Would like to know what all i should export as per need?

srinivasreddych replied:
I exported `StateMachine` Arn for now. Users can feel free to export whatever is needed as per use-case

dgraeber reviewed (commented):
Since this is a new module, another entry to gitworkflow needs to be added to to static checks

srinivasreddych commented:
> Since this is a new module, another entry to gitworkflow needs to be added to to static checks



> Since this is a new module, another entry to gitworkflow needs to be added to to static checks

I am wondering since this is an `example` pattern, do we need static checks? Should we just do them for our core-modules or everything?

a13zen commented on modules/examples/eb-sf-batch/deployspec.yaml:
```diff
@@ -0,0 +1,35 @@
+      - pip install -r requirements.txt
+    build:
+      commands:
+      - cdk destroy --force --app "python app.py"
```
Do we need to clean up ECR on destroy? Any reason we don't create ECR via CDK?

srinivasreddych replied:
Good catch. I forgot to clean up ECR on destroy. 

Also, we can create ECR repos from CDK itself, but i'd be inclined to create a repo via deployspec, because i can run the below steps in a clean order (if need be):

`Create a repo -> build an image -> scan the image (using twistlock or aws inspector or ecr scanning) -> implement custom automation to decide if i want to push to ECR or not based on vulnerability findings` and then move on to the CDK stack.

a13zen commented on modules/examples/eb-sf-batch/images/simulation-mock/src/simulation_mock/sqs_manager.py:
```diff
@@ -0,0 +1,142 @@
+    pass
+
+
+class SQSHeartbeatReceiver(SQSReceiver[ValueType]):
```
Is this used by the code or is it a leftover from the simulation example? Could we directly reference the application code for the simulation example instead of having a copy?

a13zen replied:
We could also have a super simple example batch application instead of the simulation if it reduces confusion

srinivasreddych replied:
I am not using the simulation code here, i need to cleanup. I am using a simple shell command to make sure my job ran successfully or not `Line100 of stack.py`

dgraeber commented:
> 

I think we do want to run static checks on all modules as they are committed to the repo. It sets the standard, and prevents additional work when we do actually go to use it (also, if anyone copy/paste the module, it is go-to-go).

a13zen commented on modules/examples/eb-sf-batch/images/simulation-mock/src/simulation_mock/__init__.py:
```diff
@@ -0,0 +1,35 @@
+#  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Can we delete this if we're not really using it in the module?

srinivasreddych replied:
Deleted. Thanks for identifying

a13zen commented on modules/examples/eb-sf-batch/images/simulation-mock/src/simulation_mock/simulator.py:
```diff
@@ -0,0 +1,84 @@
+#  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Can we delete this if we're not really using it in the module?

srinivasreddych replied:
Deleted. Thanks for identifying

a13zen commented on modules/examples/eb-sf-batch/modulestack.yaml:
```diff
@@ -0,0 +1,47 @@
+              - "s3:Get*"
+              - "s3:List*"
+            Effect: Allow
+            Resource: ["arn:aws:s3:::addf*", "arn:aws:s3:::cdk*"]
```
How does this limit work when bootstrapping with a different project name than addf?

srinivasreddych replied:
The example module here is to showcase the usage of the module with `addf` as the project name. customers can change it based on their naming pattern. also we encourage customers to name their projects something like `addf-clientX` so that `addf-*` will always help them

srinivasreddych commented:
Hello @a13zen Sorry i missed to address these comments as i was working on other priorities. I will refactor the above soon.

srinivasreddych commented:
@dgraeber and @a13zen : Sorry for the delay on addressing the above, pls find the below changes:
- Removed the mock simulation module which was not being used and added a sample python app which is easy to understand.
- added static checks to the module
- deleting the ECR repo upon cleanup of the module
- provided an example on how to do cdk outputs.

a13zen commented:
LGTM 👍

dgraeber commented:
This has the --skip-static-checks flag in the gitflow defintion.  We want to do the staric checks on new modules.
